### PR TITLE
[Backport 2.x] Adding Remote translog upload, download and failover changes

### PR DESCRIPTION
**Pull request by gbbafna**
### Description
https://github.com/opensearch-project/OpenSearch/pull/5755
https://github.com/opensearch-project/OpenSearch/pull/5638
https://github.com/opensearch-project/OpenSearch/pull/5392
https://github.com/opensearch-project/OpenSearch/pull/5649
https://github.com/opensearch-project/OpenSearch/pull/5699

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -43,6 +44,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Change http code for DecommissioningFailedException from 500 to 400 ([#5283](https://github.com/opensearch-project/OpenSearch/pull/5283))
 - Pre conditions check before updating weighted routing metadata ([#4955](https://github.com/opensearch-project/OpenSearch/pull/4955))
 - Support remote translog transfer for request level durability([#4480](https://github.com/opensearch-project/OpenSearch/pull/4480))
```
Comment (Bukhtawar) [inline CHANGELOG.md:46]: Lets fix the change log to include what has been added

Comment (gbbafna) [reply CHANGELOG.md:47]: Added https://github.com/opensearch-project/OpenSearch/pull/5392 for  Remote Translog Functionality

Comment (ashking94) [reply CHANGELOG.md:47]: @gbbafna  changelog from https://github.com/opensearch-project/OpenSearch/pull/5699/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed is also not there. can you add that?

Comment (sachinpkale) [reply CHANGELOG.md:47]: `Add support for refresh level durability` is added under `Added`, we can remove it from here. Also, we can remove segment store integration in failover flow entry.

**Review by Bukhtawar — APPROVED** [review]

**Pull request merged**